### ISSUE TITLE
Search by Uploader Very, Very Broken

### ISSUE THREAD

**Issue by L3Home** [bsq-ports/BetterSongSearchQuest#12]
I started noticing strange problems trying to locate songs by known uploaders, so I investigated and tested many different names of uploaders in my current search results (to ensure that the only difference in my search criteria were the uploader name(s)) and found that there are several conditions that cause Uploader searches to fail.

First, the delimiter for multiple uploaders seems to include the [Space] character. This means that any uploader with a space in their name is searched for by each part of the name, and no results appear as [I believe] while keyword searching can find partial strings, the Uploader search requires a full string match.

Also, the in-game (Quest) keyboard does not appear to have any symbols (letters and numbers only) and the search doesn't match any names that contain one (such as the [Dash] character, as seen in Dee-Dee (she has other variants, but this is one and is associated with several of her maps). It can't be typed, and using the automated button returns no results.

Of course, there's also a problem with Phoenix's tag where it contains that foreign character (don't get me started on THAT issue; it's caused problems in multiple mods). Similarly, however, users with Chinese, Korean, etc. characters cause problems.

Punctuation such as [ampersand] and/or [comma] can cause a search using the button to try to find uploaders with the name "&" and/or names with commas attached to them, etc.

Finally, even though the on-screen keyboard includes numbers, any search for an uploader who has a number in their name (like P90KILLR) either by typing or by using the button, will return no results, even when you're staring right at them in the current search list.

In short, I would recommend reworking the Uploader search; it may help to gather a list of some of the current real uploaders' names and trying them out. Anything that isn't all alphabetic and a single word (and with only one uploader, not a collaboration) should be suspect.

Thanks!

**Comment by L3Home** [comment]
Sorry, one correction! The keyboard in the Uploader name search DOES have a few symbols available. It's the song search keyboard that is lacking them.

That said, it still doesn't allow them, however! Searching for "dee-dee" (even when you're clicking the "Filter by Uploader" button and the system itself inserts "dee-dee" into the uploader search box, returns nothing. Unfortunately, unlike the song search, the Uploader search requires the name match *exactly* (which is perfectly fine) but it means many mappers can't be located. It seems the system filters out punctuation/symbols when searching, as if you search for "deedee" you get both "DeeDee" and "Dee-Dee" but if there's a "Dee Dee" you won't, because searching for any name with a space breaks it apart and only returns a mapper named "Dee." It's kind of a hit-and-miss hackabout to try to find anyone.

Also, while you can kind of hack around to find her stuff, numbers do not seem to be filtered out. They won't match but they won't be ignored either. I haven't found any way to include "P90KILLR" in the search results using the Uploader list. And since the name has to match the full string, you can't search for just "killr" like you can in the song search text box.

So while you can locate his stuff, you can only do so one-by-one, mapper by mapper, and not by using the Uploaders list the way you intended (allowing for multiple names and matching on any of them).

**Comment by FrozenAlex** [comment]
Yeah, the algo for the search by uploader is very scuffed. I'm not really sure how to fix it. The names are separated by space and minus is used for blacklisting uploaders and it breaks it as well.
